File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    i := 1
    for i < len(nums) {
        if nums[i] == nums[i-1] {
            nums = append(nums[:i], nums[i+1:]...)
        } else {
            i++
        }
    }
    return i
}

/*
func removeDuplicates(nums []int) int {
    last := nums[0]
    insertAt := 1
    for i := 1; i < len(nums); i++ {
        if nums[i] != last {
            last = nums[i]
            nums[insertAt] = last
            insertAt++
        } 
    }
    
    return insertAt
} 
*/

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize){
    int i = 1;
    int last = nums[0];
    
    for (int j = 1; j < numsSize; j++) {
        if (nums[j] != last) {
            nums[i] = nums[j];
            last = nums[j];
            i++;
        }
    }
    
    return i;
}